pallets/asset/src/benchmarking.rs

```diff
@@ -203,83 +218,68 @@ benchmarks! {
-
         assert_eq!(Module::<T>::identifiers(ticker), identifiers);
     }
 
-
     freeze {
-        let owner = UserBuilder::default().generate_did().build("owner");
-        let ticker = make_asset::<T>(&owner);
-    }: _(owner.origin, ticker.clone())
+        let (owner, ticker) = owned_ticker::<T>();
+    }: _(owner.origin, ticker)
     verify {
-        assert_eq!( Module::<T>::frozen(&ticker), true);
+        assert_eq!(Module::<T>::frozen(&ticker), true);
     }
 
     unfreeze {
-        let owner = UserBuilder::default().generate_did().build("owner");
-        let ticker = make_asset::<T>(&owner);
+        let (owner, ticker) = owned_ticker::<T>();
 
-        Module::<T>::freeze( owner.origin().into(), ticker.clone())
+        Module::<T>::freeze(owner.origin().into(), ticker)
             .expect("Asset cannot be frozen");
 
-        assert_eq!( Module::<T>::frozen(&ticker), true);
-    }: _(owner.origin, ticker.clone())
+        assert_eq!(Module::<T>::frozen(&ticker), true);
+    }: _(owner.origin, ticker)
     verify {
-        assert_eq!( Module::<T>::frozen(&ticker), false);
+        assert_eq!(Module::<T>::frozen(&ticker), false);
     }
 
     rename_asset {
         // New token name length.
         let n in 1 .. T::AssetNameMaxLength::get() as u32;
 
         let new_name = AssetName::from(vec![b'N'; n as usize].as_slice());
-        let owner = UserBuilder::default().generate_did().build("owner");
-        let ticker = make_asset::<T>(&owner);
-    }: _(owner.origin(), ticker.clone(), new_name.clone())
+        let (owner, ticker) = owned_ticker::<T>();
+    }: _(owner.origin(), ticker, new_name.clone())
     verify {
-        let token = Module::<T>::token_details(ticker);
-        assert_eq!( token.name, new_name);
+        assert_eq!(Module::<T>::token_details(ticker).name, new_name);
     }
 
     issue {
-        let owner = UserBuilder::default().generate_did().build("owner");
-        let ticker = make_asset::<T>(&owner);
-
-    }: _(owner.origin, ticker.clone(), (1_000_000 * POLY).into())
+        let (owner, ticker) = owned_ticker::<T>();
+    }: _(owner.origin, ticker, (1_000_000 * POLY).into())
     verify {
-        let token = Module::<T>::token_details(ticker);
-        assert_eq!(token.total_supply, (2_000_000 * POLY).into());
+        assert_eq!(Module::<T>::token_details(ticker).total_supply, (2_000_000 * POLY).into());
     }
 
-
     redeem {
-        let owner = UserBuilder::default().generate_did().build("owner");
-        let ticker = make_asset::<T>(&owner);
-    }: _(owner.origin, ticker.clone(), (600_000 * POLY).into())
+        let (owner, ticker) = owned_ticker::<T>();
+    }: _(owner.origin, ticker, (600_000 * POLY).into())
     verify {
-        let token = Module::<T>::token_details(ticker);
-        assert_eq!(token.total_supply, (400_000 * POLY).into());
+        assert_eq!(Module::<T>::token_details(ticker).total_supply, (400_000 * POLY).into());
```

Comment:
`Module::<T>::token_details` could be deduped by saving for ticker, field and value in a macro.

Reply:
The marginal utility of that doesn't seem too great to me :)